src/scout_apm/core/n_plus_one_call_set.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+
+import logging
+
+# Logging
+logger = logging.getLogger(__name__)
+
+
+class NPlusOneCallSet():
+    def __init__(self):
+        self.items = {}
+
+    def update(self, sql_string, call_count, call_duration):
+        self.find_for(sql_string).update(call_count, call_duration)
+
+    def should_capture_backtrace(self, sql_string):
+        return self.find_for(sql_string).should_capture_backtrace()
+
+    def find_for(self, sql_string):
+        if sql_string not in self.items:
+            self.items[sql_string] = NPlusOneCallSetItem(sql_string)
+        return self.items[sql_string]
+
+
+class NPlusOneCallSetItem():
+    # Fetch backtraces on this number of same SQL calls
+    CALL_COUNT_THRESHOLD = 5
+
+    # Minimum time in seconds before we start performing any work.
+    DURATION_THRESHOLD = 150/1000.0
+
+    def __init__(self, sql_string):
+        self.sql_string = sql_string
+        self.captured = False
+        self.call_count = 0
+        self.call_duration = 0.0  # In Seconds
+        self.past_duration_threshold = False
+
+    def update(self, call_count, call_duration):
```

Comment:
Rather than 2 positional args, can we make this kwargs so it's not possible to flip them by accident?